### PR TITLE
Add sandcats updater

### PR DESCRIPTION
**Pull request by jparyani**

### CODE REVIEW

File: src/sandstorm/run-bundle.c++ [hunk 1]
```diff
@@ -1822,6 +1825,7 @@ private:
           ", \"allowDevAccounts\":", config.allowDevAccounts ? "true" : "false",
           ", \"isTesting\":", config.isTesting ? "true" : "false",
           ", \"wildcardHost\":\"", config.wildcardHost, "\"",
+          ", \"sandcatsHostname\":\"", config.sandcatsHostname, "\"",
```
Comment (kentonv) [inline src/sandstorm/run-bundle.c++]: I guess you're relying on Javascript treating empty strings as falsy, but it seems safer to not include this field or explicitly set it null if sandcatsHostname isn't set...

File: shell/server/sandcats.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+var SANDCATS_HOSTNAME = Meteor.settings && Meteor.settings.public &&
+                          Meteor.settings.public.sandcatsHostname;
+var SANDSTORM_ALTHOME = Meteor.settings && Meteor.settings.home;
+SANDCATS_VARDIR = (SANDSTORM_ALTHOME || "") + "/var/sandcats";
+
+var ROOT_URL = Url.parse(process.env.ROOT_URL);
+var HOSTNAME = ROOT_URL.hostname;
+var SANDCATS_NAME; // Look at `startup` below to see where this is set
+
+updateSandcats = function () {
```
Comment (kentonv) [inline shell/server/sandcats.js]: Seems like these functions don't need to be global?

**Comment by jparyani** [comment]
Done

File: shell/server/sandcats.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+var querystring = Npm.require("querystring");
+var https = Npm.require("https");
+var fs = Npm.require("fs");
+var dgram = Npm.require("dgram");
+var Url = Npm.require("url");
+
+var SANDCATS_HOSTNAME = Meteor.settings && Meteor.settings.public &&
+                          Meteor.settings.public.sandcatsHostname;
+var SANDSTORM_ALTHOME = Meteor.settings && Meteor.settings.home;
+var SANDCATS_VARDIR = (SANDSTORM_ALTHOME || "") + "/var/sandcats";
```
Comment (kentonv) [inline shell/server/sandcats.js]: Actually, doesn't proxy.js already expose these as globals? I don't think you need to copy them here.

File: shell/server/sandcats.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  });
+  req.write(post_data);
+  req.end();
+
+  req.on("error", function(err) {
+    console.error("Couldn't send update sandcats hostname", err);
+  });
+};
+
+var pingUdp = function () {
```
Comment (kentonv) [inline shell/server/sandcats.js]: should probably be `function pingUdp() {`. :)

(also for `updateSandcats`)

**Comment by jparyani** [comment]
Done

**Pull request merged**